### ISSUE TITLE
Documentation Questions

### ISSUE THREAD

**Issue by martinheidegger** [tradle/KYCDeepFace#11]
Collection Issue for open questions about this repository

 - [x] How was the model for the [detector](https://github.com/tradle/KYCDeepFace/blob/69b00018fd5446f755bdff3d9fd37a6f103a037d/vision/landmark_detector.py#L18) trained? DataSource? License? → [see here](https://github.com/tradle/KYCDeepFace/issues/11#issuecomment-1035867599)
- [ ] Where does [the logic](https://github.com/tradle/KYCDeepFace/blob/69b00018fd5446f755bdff3d9fd37a6f103a037d/core/face_landmarks.py#L110-L135) for the face orientation come from? Are there papers or documentation on this approach?
- [ ] Find a good: #9 
- [ ] [Ideas to improve accuracy](https://github.com/tradle/KYCDeepFace/issues/10) ?
- [ ] Best approach to: #7
- [ ] https://github.com/tradle/KYCDeepFace/issues/12
- [x] Document the [use/function of ArcMarginProduct](https://github.com/tradle/KYCDeepFace/blob/233f33d62daa6dd4eeae9433cab58822f8208a0d/train.py#L154) in the training process. → [see here](https://github.com/tradle/KYCDeepFace/issues/11#issuecomment-1035867599)
- [ ] Does the code contain copied code from other repositories?

**Comment by urbien** [comment]
1, 2, 4, 6, 7, 8 - clearly to @fyr91 
but the rest are not

I thought the measure of accuracy is already present. Why do you think it is not good?

**Comment by martinheidegger** [comment]
The first documentation of the lfw_eval and the inference processing I put together [here](https://hackmd.io/6KMGjzDjQ0yIvcEaKhBZTw?view)

**Comment by urbien** [comment]
can you move the documents to github?

**Comment by martinheidegger** [comment]
Added to the PR #8 in commits https://github.com/tradle/KYCDeepFace/commit/74a1f4128d9bbb4ff346d84d65631ef4040e77a4  and https://github.com/tradle/KYCDeepFace/pull/8/commits/b57289e157605a1ef0f4fb5881a2c702fc5753a3

**Comment by fyr91** [comment]
> How was the model for the [detector](https://github.com/tradle/KYCDeepFace/blob/69b00018fd5446f755bdff3d9fd37a6f103a037d/vision/landmark_detector.py#L18) trained? DataSource? License?

Face detector are pre-trained model obtained from [Link](https://github.com/Linzaer/Ultra-Light-Fast-Generic-Face-Detector-1MB). Its under [MIT](https://github.com/Linzaer/Ultra-Light-Fast-Generic-Face-Detector-1MB/blob/master/LICENSE) license

> Where does [the logic](https://github.com/tradle/KYCDeepFace/blob/69b00018fd5446f755bdff3d9fd37a6f103a037d/core/face_landmarks.py#L110-L135) for the face orientation come from? Are there papers or documentation on this approach?

Face orientation is the one we will be facing during the actual usage. When people upload their image, it might not be in the correct orientation. We will use the exif data stored in the image data to determine the orientation of image and correct it accordingly.

>  [Document the structure/source of `pairs.txt` for lfw_eval.py #12](https://github.com/tradle/KYCDeepFace/issues/12)

Lfw dataset is obtained [here](http://vis-www.cs.umass.edu/lfw/lfw-funneled.tgz), within the extracted folder, you will be able to find a pair.txt file for evaluation purpose.

> Document the [use/function of ArcMarginProduct](https://github.com/tradle/KYCDeepFace/blob/233f33d62daa6dd4eeae9433cab58822f8208a0d/train.py#L154) in the training process.

For ArcMargin please refer to [arcface](https://paperswithcode.com/method/arcface).

> Does the code contain copied code from other repositories?

Yes, specifically for recognition training part, it's referred from [arcface](https://github.com/ronghuaiyang/arcface-pytorch).

**Comment by martinheidegger** [comment]
Thank you @fyr91 for filling in the blanks. I have a few follow ups:

> Face orientation .... correct it accordingly.

Thank you for explaining how it works, but I was more wondering how did got to this code? Did you think about this yourself? Did you find a paper that described that approach?

Anyways, it seems like this code will be changed to https://github.com/tradle/KYCDeepFace/pull/8#issuecomment-1023517512 which is pretty much self explanatory.

> Lfw dataset is obtained [here](http://vis-www.cs.umass.edu/lfw/lfw-funneled.tgz), within

This link returns a 403 - Forbidden?!  Is the following a mirror of the same data-set? https://drive.google.com/file/d/1p1wjaqpTh_5RHfJu4vUh8JJCdKwYMHCp/view

> Yes, specifically for recognition training part

Just to be clear, this is the only copied code?
